### PR TITLE
check if torrent is destroyed before emitting download/upload event

### LINKED ISSUE #999: Cannot read property 'emit' of null
Just so you know :) while downloading I got the following:

```
/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/webtorrent/lib/torrent.js:927
    self.client.emit('upload', uploaded)
               ^
TypeError: Cannot read property 'emit' of null
    at Wire.<anonymous> (/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/webtorrent/lib/torrent.js:927:16)
    at emitOne (events.js:96:13)
    at Wire.emit (events.js:188:7)
    at Wire.piece (/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/bittorrent-protocol/index.js:337:8)
    at respond (/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/bittorrent-protocol/index.js:498:10)
    at /Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/fs-chunk-store/index.js:200:7
    at end (/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/run-parallel/index.js:16:15)
    at done (/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/run-parallel/index.js:20:10)
    at each (/Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/run-parallel/index.js:26:7)
    at /Users/davidgatti/Documents/GitHub/MrWhite/How-to-Stream-Torrents-in-NodeJS/node_modules/run-parallel/index.js:41:37
```

This happened after playing 2 weeks with this module, meaning it isn't frequent, but it did happen :)

* WebTorrent version: 0.98.0
* Node.js version: v6.4.0
* Browser name/version (if using WebTorrent in the browser): Using NodeJS module.

### LINKED ISSUE #1924: Can't remove/destroy torrents from events
**What version of this package are you using?**
https://www.npmjs.com/package/websocket/v/1.0.32 

**What operating system, Node.js, and npm version?**
Debian 10 (buster)
Node v12.18.3 

**What happened?**
`client.remove()` and `torrent.destroy()` , while called in a torrent event, throw this error : 
```console
/some/path/node_modules/webtorrent/lib/torrent.js:917
this.client.emit('download', downloaded)
                  ^
TypeError: Cannot read property 'emit' of null
```
```js
client.add(torrentId, { path: downloadPath }, torrent = > {

  torrent.on('done', () => {
    get_the_file_and_send_it_through_websocket(connection)
  })

  torrent.on('download', dl => {
    if(socket.connected) {
      socket.send(torrent.progress)
    } else {
      client.remove(torrent.torrentFile)
      // torrent.destroy()
    }
  })

})
```
I also tried `torrent.removeAllListeners('download')` but it looks like a problem with `client` 

**What did you expect to happen?**
I'm sending the data from server to clients through websockets :
I'd like to remove/destroy torrents when a user leaves midway.

### PR DESCRIPTION
**Pull request by JiPaix**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x ] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
`Torrent.client` was emitting `download` and `upload` events without checking if `this.client` was still defined

**Which issue (if any) does this pull request address?**
This PR fixes #1924 and fixes #999 (duplicate).

### CODE REVIEW

File: lib/torrent.js [hunk 1]
```diff
@@ -921,7 +921,7 @@ class Torrent extends EventEmitter {
       this._downloadSpeed(downloaded)
       this.client._downloadSpeed(downloaded)
       this.emit('download', downloaded)
-      this.client.emit('download', downloaded)
+      if (this.client) this.client.emit('download', downloaded)
```
Comment (alxhotel) [inline lib/torrent.js:924]: I would rather use:
```suggestion
      if (this.destroyed) return
      this.client.emit('download', downloaded)
```
Just to be consistent with the rest of the code. (same thing with the other change)

Comment (JiPaix) [reply lib/torrent.js]: suggestion applied with https://github.com/webtorrent/webtorrent/pull/1925/commits/4a5597149541709a1a95bcd46bfbb7efcf3b4102

**Review by alxhotel — APPROVED** [review]
Thanks, @JiPaix!!

LGTM

**Review by DiegoRBaquero — APPROVED** [review]

**Comment by feross** [comment]
Great to finally have this bug fixed!

**Pull request merged**